app/controllers/reports/activity_graph_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Reports::ActivityGraphController < ApplicationController
+  def index
+    setup_date_range_picker
+    @distribution_data = received_distributed_data(helpers.selected_range)
+  end
+
+  private
+
+  def total_purchased_unformatted(range = selected_range)
+    LineItem.active.where(itemizable: current_organization.purchases.during(range)).sum(:quantity)
+  end
+
+  def total_distributed_unformatted(range = selected_range)
+    LineItem.active.where(itemizable: current_organization.distributions.during(range)).sum(:quantity)
+  end
+
+  def total_received_donations_unformatted(range = selected_range)
+    LineItem.active.where(itemizable: current_organization.donations.during(range)).sum(:quantity)
+  end
+
+  def received_distributed_data(range = selected_range)
```

Comment:
All these methods seem like they should be part of a report/service, not controller methods.

Reply:
... and it looks like they are used from helpers. I don't like how helpers aren't really namespaced -- I think the existing ones are on the dashboard helpers still but should be moved into individual page helpers :(

Reply:
These are all now moved into app/controllers/reports_controller.rb; they are used to build a dataset passed into the view for the given date range. I think they're fine there, but can move them some place specific if you have something else in mind.

Reply:
Yeah I'd rather not have these queries in a controller as they're totally unrelated to rendering content. Can we create a module like `DashboardQueries` we can use for this and any future queries?

Reply:
Actually... if you look at a file like `View::Inventory`, I think an object like that probably works better for what we want here. Essentially we can incorporate all the queries into a single object (or set of objects) that can be passed down in the view. Even doing things like `sum` and `count` in the view makes me uncomfortable... I much prefer views to be extremely dumb and doing nothing but passing existing data into (possibly) rendering functions like `number_to_delimiter`.

Having a view object which we can calculate allows us a) to not have to deal with five or more instance variables and just have one, and b) to centralize all the logic that calculates that one view object.

Reply:
It also makes it much easier to test the query logic separately from the view logic... right now it's hard to do that without a big meaty request spec.